BOARD: clean naming of profiling data *.gcda

Apaczer opened the pull request:
Otherwise it brakes MinGW git or be it WSL2 with native git - to long file names for Win FS.

Apaczer commented:
welp the compiler ask for `#home#runner#work#buildroot#buildroot#output#build#libretro-gpsp-092c5144827ee3ac07a6af9276244221c303ca41#<src_file>.gcda` so need to find a workaround to explictly point to only `<src_file>.gcda` or maybe run prebuild script and change the naming on the run. 

Anyway you can't use normally provided *.gcda data in current state because for e.g. when running locally on your machine, a working directory of compiler changes.

Apaczer committed:
[BOARD] don't use "preserve-paths" in srcname of profiling *.gcda data

for file in *; do cleanname=$(echo "$file" | sed 's/^.*<commit>#//') ; mv "$file" "$cleanname" ; done

for file in *; do newdirectory=$(echo "$file" | sed 's/#[^#]*$//' | sed 's/#/\//g') ; mkdir -p "$newdirectory" ; newname=$(echo "$file" | sed 's/^.*#//') ; mv "$file" "$newdirectory/$newname" ; done

Apaczer committed:
[PACKAGE] update PGO optimized libretro cores to not use `-fprofile-dir=path`

TriForceX commented:
> Otherwise it brakes MinGW git or be it WSL2 with native git - to long file names for Win FS.

mm ive had similar problems with paths so i solved it by using `git config --system core.longpaths true` i wonder if this affects also files (or if there is a counterpart for files)

Apaczer committed:
[PACKAGE] optionize PGO optimized builds

PROFILE=
0 - don't use PGO
YES - generate data at runtime
APPLY - apply optimization at compilation step

Apaczer commented:
If you want directory structure intact in `profile/` data output don't indicate path in `-fprofile-generate`.

or run this script for output .gcda files (replace \<commit\> with git hash):
```
for file in *; do cleanname=$(echo "$file" | sed 's/^.*<commit>#//') ; mv "$file" "$cleanname" ; done

for file in *; do newdirectory=$(echo "$file" | sed 's/#[^#]*$//' | sed 's/#/\//g') ; mkdir -p "$newdirectory" ; newname=$(echo "$file" | sed 's/^.*#//') ; mv "$file" "$newdirectory/$newname" ; done
```
